http/middleware_test.go

```diff
@@ -38,6 +56,19 @@ func TestWithHeaderAuthorization_InvalidAuthorization(t *testing.T) {
 	res, err = ts.Client().Do(req)
 	require.NoError(t, err)
 	require.Equal(t, http.StatusOK, res.StatusCode)
```

Comment:
❓ Is this correct? Shouldn't this case return 401 too?

Reply:
Back when I first implemented the middleware, I decided to keep the functionality that v1 executes when the bearer token cannot be decoded.

https://github.com/clerk/clerk-sdk-go/blob/59c1229871a66eebbc8bba4cc0855a3da1357e6f/clerk/middleware_v2.go#L60

https://github.com/clerk/clerk-sdk-go/blob/195ed8614b6a20fde4cdb95ff51bccce06976ddb/http/middleware.go#L60

That's why I also added the test case, because it was not intuitive to me as well.

Reply:
Got it, thanks for clarifying!